opentelemetry-appender-tracing/src/layer.rs

```diff
@@ -99,23 +167,21 @@ where
         event: &tracing::Event<'_>,
         _ctx: tracing_subscriber::layer::Context<'_, S>,
     ) {
-        let meta = event.metadata();
+        let normalized_meta = event.normalized_metadata();
+        let meta = normalized_meta.as_ref().unwrap_or_else(|| event.metadata());
```

Comment:
Can we run the stress test to see if this is adding to any cost - probably compare with and without changes?

Reply:
Seems to be this one https://github.com/open-telemetry/opentelemetry-rust/blob/main/stress/README.md :)

Reply:
yes, this is for logs - https://github.com/open-telemetry/opentelemetry-rust/blob/main/stress/src/logs.rs

Reply:
Interesting.

So on my machine, here is the result :

- Main : ~ 6.2M iterations/sec
- On my branch : ~ 6.3M iterations/sec (a little bit faster)
- With experimental attributes : ~ 4.7M iterations/sec (24% slower)

However, by modifying the code, I found the is_metadata function is really quick and has a very minimal impact on performance. (~6.3M vs ~6.2M).

It is adding the metadata attributes that makes the logger slower. Probably allocating memory that slow down the whole thing.

Reply:
On Main, if I remove all attributes in the event in the test_log function, the performance goes up to 8M iterations/sec.

Reply:
@cijothomas Where are the basic criterion benchmarks for metrics and traces?

Reply:
https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry-sdk/benches/trace.rs
https://github.com/open-telemetry/opentelemetry-rust/blob/main/opentelemetry-sdk/benches/metric_counter.rs

Reply:
Thanks! Do you think it is better to do the new benches in a new PR? I think it would be easier to compare the performance before and after this PR.

Reply:
> Do you think it is better to do the new benches in a new PR?

Yes!